src/shared/condition.c

```diff
@@ -958,6 +960,38 @@ static int condition_test_file_is_executable(Condition *c, char **env) {
                 (st.st_mode & 0111));
 }
 
+static int condition_test_psi(Condition *c, char **env) {
+        ResourcePressure pressure;
+        PressureType type;
+        loadavg_t limit;
+        char *path;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(IN_SET(c->type, CONDITION_MEMORY_PRESSURE, CONDITION_CPU_PRESSURE));
+
+        if (!is_pressure_supported())
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Pressure Stall Information (PSI) is not supported");
+
+        type = c->type == CONDITION_CPU_PRESSURE ? PRESSURE_TYPE_SOME : PRESSURE_TYPE_FULL;
+        path = strjoina("/proc/pressure/", c->type == CONDITION_CPU_PRESSURE ? "cpu" : "memory");
+
+        r = read_resource_pressure(path, type, &pressure);
+        if (r < 0)
+                return log_debug_errno(r, "Error parsing pressure from %s: %m", path);
+
+        r = parse_permyriad(c->parameter);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse permyriad: %s", c->parameter);
+
+        r = store_loadavg_fixed_point((unsigned long) r / 100, (unsigned long) r % 100, &limit);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse loadavg: %s", c->parameter);
+
+        return pressure.avg300 < limit;
```

Comment:
shouldn't this all be "below or equal" rather than "below"?

Reply:
I guess it depends on how we document it? The manpage says "below" atm